code/obj/item/mob_parts/robot_parts.dm

```diff
@@ -1216,6 +1212,257 @@ ABSTRACT_TYPE(/obj/item/parts/robot_parts/leg/right)
+				boutput(user, "<span class='alert'>You need to move closer!</span>")
+				return
+
+			switch(action)
+				if("Finish and Activate the Drone")
+					src.finish_drone()
+				if("Remove the AI Interface")
+					src.ai_interface.set_loc( get_turf(src) )
+					src.ai_interface = null
+					src.UpdateIcon()
+				if("Remove the Power cell")
+					src.cell.set_loc( get_turf(src) )
+					src.cell = null
+					src.UpdateIcon()
+			playsound(src, 'sound/items/Ratchet.ogg', 40, 1)
+			src.UpdateIcon()
+			return
+
+	update_icon()
+		src.icon_state = "frame-" + src.shelltypetoapply;
+
+		if(build_step > 0)
+			src.UpdateOverlays(image('icons/mob/hivebot.dmi', "plating-c-" + src.shelltypetoapply, FLOAT_LAYER, 2),"plating")
+		else
+			src.UpdateOverlays(null,"plating")
+
+		if(build_step > 1)
+			src.UpdateOverlays(image('icons/mob/hivebot.dmi', "wires-c-" + src.shelltypetoapply, FLOAT_LAYER, 2),"wires")
+		else
+			src.UpdateOverlays(null,"wires")
+
+		if(build_step > 2)
+			src.UpdateOverlays(image('icons/mob/hivebot.dmi', "radio-c-" + src.shelltypetoapply, FLOAT_LAYER, 3),"radio")
+		else
+			src.UpdateOverlays(null,"radio")
+
+		if(src.ai_interface)
+			src.UpdateOverlays(image('icons/mob/hivebot.dmi', "interface-c-" + src.shelltypetoapply, FLOAT_LAYER, 3),"interface")
+		else
+			src.UpdateOverlays(null,"interface")
+
+		if(src.cell)
+			src.UpdateOverlays(image('icons/mob/hivebot.dmi', "cell-c-" + src.shelltypetoapply, FLOAT_LAYER, 4),"cell")
+		else
+			src.UpdateOverlays(null,"cell")
+
+
+	proc/check_completion()
+		if (build_step == 3)
+			if ((src.ai_interface) && src.cell)
+				return 1
+		return 0
+
+	proc/finish_drone()
+		var/mob/living/silicon/drone/drone = null
+		drone = new /mob/living/silicon/drone(get_turf(src.loc),src,0,src.emagged)
+
+		drone.name = "Drone"
+		drone.real_name = "Drone"
+
+		if (src.ai_interface)
+			if (!(drone in available_ai_shells))
+				available_ai_shells += drone
+			for_by_tcl(AI, /mob/living/silicon/ai)
+				boutput(AI, "<span class='success'>[src] has been connected to you as a controllable drone.</span>")
+			playsound(src, 'sound/weapons/radxbow.ogg', 40, 1)
+			drone.ai_interface = src.ai_interface
+			drone.ai_interface.set_loc(drone)
+			drone.shell = 1
+		else
+			stack_trace("We finished drone [identify_object(drone)] from frame [identify_object(src)] with a brain, but somehow lost the brain??? Where did it go")
+			drone.death()
+			qdel(src)
+			return
+
+		if (src.cell)
+			drone.cell = src.cell
+			drone.cell.set_loc(drone)
+
+		if (src.moduletoapply)
+			drone.module = (moduletoapply)
+			drone.module.set_loc(drone)
+			drone.hud.update_module()
+			drone.hud.module_added()
+
+		drone.update_appearance()
+
+		qdel(src)
+		return
+
+/obj/item/parts/robot_parts/drone_frame/complete_shell_choosable
+	build_step = 3
+	cell = new/obj/item/cell/shell_cell/charged
+	ai_interface = new/obj/item/ai_interface
+	New()
+		..()
+		src.finish_drone()
```

Comment:
If these are immediately turning into drones, what's the use case for this where you wouldn't just instantiate a drone?

Reply:
I tried that at first! Creating the drone directly had some difficulties with... interface boards being unremovable, the HUD not showing up, a few other things... I couldn't figure out why for the life of me, so I pulled this as a workaround because it didn't cause those same issues.
I could try taking a further look at it, but I really have not the slightest what was going wrong with it, hwhaha.